controllers/datamovement_controller_test.go

```diff
@@ -91,7 +197,288 @@ var _ = Describe("Data Movement Test" /*Ordered, (Ginkgo v2)*/, func() {
+			Eventually(func(g Gomega) string {
+				cmd := ""
+				g.Expect(k8sClient.Get(context.TODO(), client.ObjectKeyFromObject(dm), dm)).To(Succeed())
+				if dm.Status.CommandStatus != nil {
+					cmd = dm.Status.CommandStatus.Command
+				}
+				return cmd
+			}).Should(Equal(fmt.Sprintf(
+				"mpirun --allow-run-as-root -np 1 --host localhost dcp --progress 1 %s %s", srcPath, destPath)))
+		})
+	})
+
+	Context("when the dm config map has specified a valid dmProgressInterval", func() {
+		BeforeEach(func() {
+			setDefaultConfigValues()
+			cmData[DMConfigKeyProgInterval] = "25s"
+		})
+
+		It("parseConfigMapValues should return that value", func() {
+			configMap := &corev1.ConfigMap{}
+			Expect(k8sClient.Get(ctx, types.NamespacedName{Name: DMConfigMapName, Namespace: DMConfigMapNamespace}, configMap)).To(Succeed())
+			Expect(configMap.GetLabels()[testLabelKey]).To(Equal(testLabel))
+			_, dmProgressInterval, _ := parseConfigMapValues(*configMap, 1)
+			Expect(dmProgressInterval).To(Equal(25 * time.Second))
+		})
+	})
+
+	Context("when the dm config map has specified a invalid dmProgressInterval", func() {
+		BeforeEach(func() {
+			setDefaultConfigValues()
+			cmData[DMConfigKeyProgInterval] = "aafdafdsa"
+		})
+
+		It("parseConfigMapValues should return the default value", func() {
+			configMap := &corev1.ConfigMap{}
+			Expect(k8sClient.Get(ctx, types.NamespacedName{Name: DMConfigMapName, Namespace: DMConfigMapNamespace}, configMap)).To(Succeed())
+			Expect(configMap.GetLabels()[testLabelKey]).To(Equal(testLabel))
+			_, dmProgressInterval, _ := parseConfigMapValues(*configMap, 1)
+			Expect(dmProgressInterval).To(Equal(DMConfigDefaultProgInterval))
+		})
+	})
+
+	Context("when the dm config map has specified a empty dmProgressInterval", func() {
+		BeforeEach(func() {
+			setDefaultConfigValues()
+			cmData[DMConfigKeyProgInterval] = ""
+		})
+
+		It("parseConfigMapValues should return the default value", func() {
+			configMap := &corev1.ConfigMap{}
+			Expect(k8sClient.Get(ctx, types.NamespacedName{Name: DMConfigMapName, Namespace: DMConfigMapNamespace}, configMap)).To(Succeed())
+			Expect(configMap.GetLabels()[testLabelKey]).To(Equal(testLabel))
+			_, dmProgressInterval, _ := parseConfigMapValues(*configMap, 1)
+			Expect(dmProgressInterval).To(Equal(DMConfigDefaultProgInterval))
+		})
+	})
+
+	Context("when there is no dm config map", func() {
+		BeforeEach(func() {
+			setDefaultConfigValues()
+			createCm = false
+		})
+
+		It("should requeue and not start running", func() {
+			Consistently(func(g Gomega) string {
+				g.Expect(k8sClient.Get(context.TODO(), client.ObjectKeyFromObject(dm), dm)).To(Succeed())
+				return dm.Status.State
+			}).ShouldNot(Equal(nnfv1alpha1.DataMovementConditionTypeRunning))
+		})
+	})
+
+	Context("when a data movement command fails", func() {
+		BeforeEach(func() {
+			setDefaultConfigValues()
+			cmData[DMConfigKeyCmd] = "" // try to use mpirun dcp, it will fail
```

Comment:
Would running "false" be more obvious than a botched mpirun?

Reply:
I've changed it to false.